autodepgraph/graph.py

```diff
@@ -1,33 +1,71 @@
 from qcodes.instrument.base import Instrument
 from qcodes.instrument.parameter import ManualParameter
 from qcodes.utils import validators as vals
-"""
-This contains the definition
-"""
+import yaml
+import logging
 
 
-class graph():
+class Graph(Instrument):
     """
     A class containing nodes
     """
 
-    def load_graph(self, filename):
+    def __init__(self, name):
+        self.name = name
+
+        self._nodes = {}
+
+    def load_graph(self, filename, load_node_state=False):
         """
         Loads a graph.
         """
-        raise NotImplementedError()
+        graph_snap = yaml.load(open(filename, 'r'))
+        for key, node_snap in graph_snap['nodes'].items():
+            try:
+                # Look for an existing node
+                node = self.find_instrument(node_snap['name'])
+                # no parameters indicates closed instrument -> open a new one
+                if not hasattr(node, 'parameters'):
+                    node = CalibrationNode(node_snap['name'])
+            except KeyError:
+                # If the node does not exist, create a new node
+                node = CalibrationNode(node_snap['name'])
+            # these are the directed edges of each node
```

Comment:
'these' in this comment refers to the dependencies, I think. But why is this comment here, I don't see the context.